c/src/user_manager.rs

```diff
@@ -26,82 +26,68 @@ use super::{
-// /// @param username The name of the user to set the password of
-// /// @param password The new password
-// #[no_mangle]
-// pub extern "C" fn users_set_password(
-//     user_manager: *const UserManager,
-//     username: *const c_char,
-//     password: *const c_char,
-// ) {
-//     unwrap_void(borrow(user_manager).set_password(string_view(username), string_view(password)));
-// }
+/// Retrieves the username of the user who opened this connection
+#[no_mangle]
+pub extern "C" fn users_current_username(driver: *const TypeDBDriver) -> *mut c_char {
+    release_string(borrow(driver).users().current_username().clone().to_owned())
+}
+
+/// Iterator over a set of <code>User</code>s
+pub struct UserIterator(CIterator<User>);
+
+/// Forwards the <code>UserIterator</code> and returns the next <code>User</code> if it exists,
+/// or null if there are no more elements.
+#[no_mangle]
+pub extern "C" fn user_iterator_next(it: *mut UserIterator) -> *mut User {
+    unsafe { iterator_next(addr_of_mut!((*it).0)) }
+}
+
+/// Frees the native rust <code>UserIterator</code> object
+#[no_mangle]
+pub extern "C" fn user_iterator_drop(it: *mut UserIterator) {
+    free(it);
+}
+
+/// Retrieves all users which exist on the TypeDB server.
+#[no_mangle]
+pub extern "C" fn users_all(driver: *const TypeDBDriver) -> *mut UserIterator {
+    try_release(borrow(driver).users().all().map(|users| UserIterator(CIterator(box_stream(users.into_iter())))))
+}
+
+/// Checks if a user with the given name exists.
+#[no_mangle]
+pub extern "C" fn users_contains(driver: *const TypeDBDriver, username: *const c_char) -> bool {
+    unwrap_or_default(borrow(driver).users().contains(string_view(username)))
+}
+
+/// Creates a user with the given name &amp; password.
+#[no_mangle]
+pub extern "C" fn users_create(driver: *const TypeDBDriver, username: *const c_char, password: *const c_char) {
+    unwrap_void(borrow(driver).users().create(string_view(username), string_view(password)));
+}
+
+/// Deletes the user with the given username.
+#[no_mangle]
+pub extern "C" fn users_delete(driver: *const TypeDBDriver, username: *const c_char) {
+    unwrap_void(borrow(driver).users().delete(string_view(username)));
+}
+
+/// Retrieves a user with the given name.
+#[no_mangle]
+pub extern "C" fn users_get(driver: *const TypeDBDriver, username: *const c_char) -> *mut User {
+    try_release_optional(borrow(driver).users().get(string_view(username)).transpose())
+}
+
+/// Sets a new password for a user. This operation can only be performed by administrators.
+///
+/// @param user_manager The </code>UserManager</code> object to be used.
+///                     This must be on a connection opened by an administrator.
+/// @param username The name of the user to set the password of
+/// @param password The new password
+#[no_mangle]
+pub extern "C" fn users_set_password(driver: *const TypeDBDriver, username: *const c_char, password: *const c_char) {
+    todo!("User update functions")
```

Comment:
@lolski You'll need to insert some code here, can ask Dmitrii if needed. But examples and the commented out code should be enough